### PR TITLE
Add support for Youtube uploading

### PR DESCRIPTION
**Pull request by SupSuper**
Solves issue #976. No external libraries were used. Requires the YouTube Data API v3 to be enabled in the Google Developers Console for the app.

Since there are no "video uploaders", I added it as a `FileUploader` with a `IsVideoFile` check.

To avoid code duplication, I created the following wrappers:
- `GoogleOAuth2`: Generalized Google services authentication, can be used in other uploaders. To make this work I had to expose `Uploader.SendRequestMultiPart` (normally `protected`), the only other way would be making `GoogleOAuth2` inherit `Uploader` which was kinda ugly.
- `OAuth2*` methods in `UploadersConfigFormHelper`: Methods for configuring general `IOAuth2` services that don't require custom setup, can be used in other uploaders.
If they all look good I can apply them to other uploaders.

Example upload:
https://i.imgur.com/xzGuQQr.png
https://i.imgur.com/B8rG0nw.png
https://youtu.be/tariUF7SCQU

### CODE REVIEW

**Comment by Torubloo** [comment]
Please change `Youtube` to `YouTube`

File: ShareX.UploadersLib/Enums.cs [hunk 1]
```diff
@@ -139,6 +139,8 @@ public enum FileDestination
         Transfersh,
         [Description("Plik")]
         Plik,
+        [Description("Youtube")]
```
Comment (matthewburnett) [inline ShareX.UploadersLib/Enums.cs]: Canonical spelling is "YouTube".

**Comment by Jaex** [comment]
These diffs are problem unfortunately :(

**Comment by SupSuper** [comment]
@Torubloo @Jaex Fixed.

**Comment by Jaex** [comment]
Thanks and good job :)

**Comment by Hilarioma** [comment]
Sorry if I am not doing things correctly. I see this thread "closed" and don't know if I should ask this in another place.
From what I am reading, you have integrated YOUTUBE uploader. MARVELLOUS!!! Godd work and thanks very much.
But I don't see it on the last version, then I don't know if I have to do something  to download this functionality or if there are till steps to do before been able to use it.
I am really a dumb with the protocol to follow on github. ;-(
thanks for any clarification.

**Comment by SupSuper** [comment]
@Hilarioma It was just added to source, it will be in the next version.

**Pull request merged**